File: labs/06/Average.java
/* Average.java is a driver for method avg()
 *
 * Input: array, an array of doubles AND size, a double
 * Precondition: array, is an array of doubles
 * Output: the average of the doubles in array
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College
 * Completed by: Ian Christensen, for CS 214 at Calvin College
 * Date: Spring, 2018
 */

import java.io.*;

public class Average {
  public static void main(String[] args) {
    double[] array0 = {};
    double[] array1 = {9.0, 8.0, 7.0, 6.0};
    System.out.println("The first average is " + avg(array0));
    System.out.println("The second average is " + avg(array1));
  }
  
  /* Function: computes the sum of an array of doubles
   * Parameters: array, an array of doubles
   * Preconditions: array, is an array of doubles
   * Returns: the sum of the values in array
   */
  public static double sum(double[] array){ 
      double total = 0.0;
      for (int i = 0; i < array.length; i++){
          total += array[i];
      }
      return total;
  }
  
  /* Function: computes the average of an array of doubles
   * Parameters: array, an array of doubles
   * Preconditions: array, is an array of doubles
   * Returns: the average of the values in array
   */
   public static double avg(double array[]) {
    if (array == null || array.length <= 0)
      return 0.0;
    else
      return sum(array) / array.length;
   }
}

File: labs/06/average.rb
#!/usr/bin/ruby
# average.rb averages the values of an array of doubles
#
# Input: array, an array of doubles AND size, a double
# Precondition: array, is an array of numbers
# Output: the average of the values in array
#
# Begun by: Dr. Adams, for CS 214 at Calvin College
# Completed by: Ian Christensen, for CS 214 at Calvin College
# Date: Spring, 2018

# Function: computes the sum of an array of doubles
# Parameters: array, an array of doubles
# Preconditions: array, is an array of doubles
# Returns: the sum of the values in array
def sum(array)
	total = 0
	array.each do |item|
		total += item
	end
	total
end

# Function: computes the average of an array of doubles
# Parameters: array, an array of doubles
# Preconditions: array, is an array of doubles
# Returns: the average of the values in array
def average(array)
	if array.empty?
		return 0.0
	else
		return sum(array) / array.size()
	end
end

# Statement: main driver for average.rb
# Parameters: none
# Precondition: average.rb is the file being run
# Returns: formatted print statements testing sum and average
if __FILE__ == $0
array0 = []
array1 = [9.0, 8.0, 7.0, 6.0]
puts "sum 0 is: #{sum(array0)}\n"
puts "sum 1 is: #{sum(array1)}\n"
puts "average 0 is: #{average(array0)}\n"
puts "average 1 is: #{average(array1)}\n"
end
